src/ApiService/ApiService/onefuzzlib/orm/EntityConverter.cs

```diff
@@ -285,7 +288,25 @@ public class EntityConverter {
                 var outputType = ef.type;
                 if (ef.discriminator != null) {
                     var (attr, typeProvider) = ef.discriminator.Value;
-                    var v = GetFieldValue(info, attr.FieldName, entity) ?? throw new Exception($"No value for {attr.FieldName}");
+                    if (iterationCount > MAX_DESERIALIZATION_RECURSION_DEPTH) {
+                        var tags = GenerateTableEntityTags(entity);
+                        tags.AddRange(new (string, string)[] {
+                            ("outputType", outputType?.Name ?? string.Empty),
+                            ("fieldName", fieldName)
+                        });
+                        _logTracer.WithTags(tags).Error($"Too many iterations deserializing {info.type}");
```

Comment:
do we need to pass a logger here to this class. We already throw an exception when we have an issue

Reply:
Instead you can use a more specific exception type if you want to collect more information about the failure. This will reduce the coupling in general

Reply:
I understand the coupling concern. My thinking was we risk losing that information if downstream callers forget to log it -- like if we do `catch (Exception)`.

Reply:
We already log exceptions when they occur at this level right. You were able to identify the issue from the logs already. 
```
Unable to extract properties from TableEntity for Microsoft.OneFuzz.Service.Task Unable to get value for property 'Config' (entity field 'config') Exception of type 'System.OutOfMemoryException' was thrown.
```

In this case if we had a more specific exception type you would get all the information you need in the logs.

Reply:
Updated to remove logtracer from entityconverter and include all the necessary information in the exception.